src/docs/sdk/performance/dynamic-sampling-context.mdx

```diff
@@ -0,0 +1,124 @@
+2. Propagating DSC to downstream SDKs.
+3. Sending the DSC to Sentry via an envelope.
+
+Because there are quite a few things to keep in mind for DSC propagation and to avoid every SDK running into the same problems, we defined a [unified propagation mechanism](#unified-propagation-mechanism) (step-by-step instructions) that all SDK implementations should be able to follow.
+
+## Baggage
+
+We chose `baggage` as the propagation mechanism for DSC. ([w3c baggage spec](https://www.w3.org/TR/baggage/))
+Baggage is a standard HTTP header with URI encoded key-value pairs.
+
+For the propagation of DSC, SDKs first read the DSC from the baggage header of incoming requests/messages.
+To propagate DSC to downstream SDKs/services, we create a baggage header (or modify an existing one) through HTTP request instrumentation.
+
+<Alert level="info">
+
+Other vendors might also be using the `baggage` header.
+If a `baggage` header already exists on an outgoing request, SDKs should aim to be good citizens by only **appending** Sentry values to the header.
+In the case that another vendor added Sentry values to an outgoing request, SDKs may overwrite those values.
+
+SDKs must not add other vendors' baggage from incoming requests to outgoing requests.
+Sentry SDKs only concern themselves with Sentry baggage.
+
+</Alert>
+
+The following is an example of what a baggage header containing Dynamic Sampling Context may look like:
+
+```
+baggage: other-vendor-value-1=foo;bar;baz, sentry-traceid=771a43a4192642f0b136d5159a501700, sentry-publickey=49d0f7386ad645858ae85020e393bef3; sentry-userid=Am%C3%A9lie, other-vendor-value-2=foo;bar;
+```
+
+See the [Payloads section](#payloads) for a complete list of key-value pairs that SDKs should propagate.
+
+## Payloads
+
+Dynamic Sampling Context is propagated via a baggage header and sent to Sentry via <Link to="/sdk/envelopes/#transaction">transaction envelope headers</Link>.
+
+### Baggage-Header
+
+SDKs may set the following key-value pairs on baggage headers.
+While all of these values are optional, SDKs should make their best effort to add as many of them to the baggage header as possible when starting a trace.
+
+- `sentry-traceid` - The original trace ID as generated by the SDK
+- `sentry-publickey` - Public key as defined by the user via the DSN in the SDK options
+- `sentry-release` - The release as defined by the user in the SDK options
+- `sentry-environment` - The environment as defined by the user in the SDK options
+- `sentry-transaction` - The name of the trace's origin transaction in unparameterized (raw) format
+- `sentry-userid` - User ID as set by the user with <Link to="/sdk/unified-api/#scope">`scope.set_user`</Link>
+- `sentry-usersegment` - User segment as set by the user with <Link to="/sdk/unified-api/#scope">`scope.set_user`</Link>
+- `sentry-samplerate` - Sample rate as defined by the user in the SDK options
```

Comment:
How should the sample rate be formatted? Do we have a max number of decimal points?

Reply:
Good question, in the JS SDK, we're currently just calling `toString` on the sample rate number which caps it per default at 16 decimal points. Happy to change it if we decide to handle this uniformly across SDKs